app/src/main/java/com/example/faunaapp/repository/Repository.java

```diff
@@ -26,25 +30,24 @@ public static Repository getInstance() {
     }
 
 
-
-
-
-
     // log in
     public void logIn(String email, String password) {
-        new LogInAsyncTask().doInBackground(email, password);
+        logInAsyncTask.doInBackground(email, password);
     }
-    public String getToken(){
-        return LogInAsyncTask.getToken();
+
+    public synchronized String getToken() {
+        String token;
+        token = logInAsyncTask.getToken();
+
+        return token;
     }
 
 
-    private static class LogInAsyncTask extends AsyncTask<String, Void, Void>
-    {
+    private static class LogInAsyncTask extends AsyncTask<String, Void, Void> {
         private static String token = "";
+
         @Override
-        protected Void doInBackground(String... strings)
-        {
```

Comment:
Isen't this interface deprecated by google? They recommend using java.util.concurrent according to the documentation. If it works, keep it, but for future development, dont implement new functionality using deprecated library functions.

Reply:
Yes, it is deprecated, but that is the way teachers taught us. But of course, later I could refactor it

Reply:
Alright. Not up to you then. 👍